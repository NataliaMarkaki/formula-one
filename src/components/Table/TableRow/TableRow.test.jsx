import * as React from "react";
import { act } from "react-dom/test-utils";
import { render, unmountComponentAtNode } from "react-dom";

describe("TableRow", () => {
  let TableRow;
  let container = null;

  beforeAll(() => {
    TableRow = require(".").default;
  });

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    jest.clearAllMocks();
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("should not render as an odd row", () => {
    act(() => {
      render(
        <TableRow
          name='Alexander'
          permanentNumber='23'
          nationality='Thai'
          DOB='1996-03-23'
          isOdd={false}
        />,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });

  it("should render as an odd row", () => {
    act(() => {
      render(
        <TableRow
          name='Alexander'
          permanentNumber='23'
          nationality='Thai'
          DOB='1996-03-23'
          isOdd={true}
        />,
        container
      );
    });
    expect(container).toMatchSnapshot();
  });
});
