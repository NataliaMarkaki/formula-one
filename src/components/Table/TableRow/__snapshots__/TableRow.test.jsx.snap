// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should not render as an odd row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 20px;
  border-top: 1px solid #dbe1e4;
}

.c1 {
  padding: 0.75em 0.75em 0.5em 0px;
  margin-left: 0px;
  width: 34%;
  vertical-align: top;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Alexander
    </div>
    <div
      class="c1"
    >
      23
    </div>
    <div
      class="c1"
    >
      Thai
    </div>
    <div
      class="c1"
    >
      1996-03-23
    </div>
  </div>
</div>
`;

exports[`TableRow should render as an odd row 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 20px;
  border-top: 1px solid #dbe1e4;
  background-color: #f6f8f9;
}

.c1 {
  padding: 0.75em 0.75em 0.5em 0px;
  margin-left: 0px;
  width: 34%;
  vertical-align: top;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      Alexander
    </div>
    <div
      class="c1"
    >
      23
    </div>
    <div
      class="c1"
    >
      Thai
    </div>
    <div
      class="c1"
    >
      1996-03-23
    </div>
  </div>
</div>
`;
